internal/storage/bucket_handle.go

```diff
@@ -250,14 +250,15 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 	pi.Token = req.ContinuationToken
 	var list gcs.Listing
 
+	var resultsCount int = 0
 	// Iterating through all the objects in the bucket and one by one adding them to the list.
 	for {
 		var attrs *storage.ObjectAttrs
 		// itr.next returns all the objects present in the bucket. Hence adding a
 		// check to break after required number of objects are returned.
 		// If req.MaxResults is 0, then wait till iterator is done. This is similar
 		// to https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/github.com/jacobsa/gcloud/gcs/bucket.go#L164
-		if req.MaxResults != 0 && (len(list.Objects) == req.MaxResults) {
```

Comment:
Do we need to maintain this extra variable?
Can't we change the check (len(list.objects) + len(list.CollapsedRuns)) == req.MaxResults?

Reply:
I don't have a strong preference on this. 
Computation wise, I think both the options are similar ( len(list.objects) + len(list.CollapsedRuns) might be little less optimised because it has extra step of fetching length )
Memory wise also, I believe both the cases are similar ? One has defined variable and other will take space to store  len(list.objects) + len(list.CollapsedRuns) in stack.
Readability wise, I prefer resultsCount. 

Please let me know if I am missing any point here.